src/main/java/de/tum/in/www1/artemis/service/ExerciseDateService.java

```diff
@@ -146,4 +146,33 @@ public static Optional<ZonedDateTime> getDueDate(ParticipationInterface particip
             return Optional.ofNullable(exercise.getDueDate());
         }
     }
+
+    /**
+     * Checks if the current time is before the assessment due date
+     * and no manual results should be published to the student.
+     *
+     * Returns false if the assessment due date is null.
+     *
+     * @param exercise to check the assessment due date
+     * @return true if the assessment due date is in the future
+     */
+    public static boolean isBeforeAssessmentDueDate(Exercise exercise) {
```

Comment:
It would be nice if you could document the behavior for equal dates.

Reply:
Since the internal comparison works on a nanosecond level and we are comparing with `ZonedDateTime.now()` (and not two saved date values) I don't think equal dates are a case to reasonably consider here. It's only important when comparing two saved values since these get truncated.

Reply:
I agree with Maximilian, though, these are bad semantics.

IsBefore is not the negation of IsAfter - NotIsBefore.

Apart from the fact that this wrapper method is rather useless. Why don't the calling methods just use the negation directly?

Reply:
I understand your point that `!before` and `after` are not completely the same, however I don't think that this wrapper is not useful.

When using the method, we'd still just do `!isBeforeAssessmentDueDate()` anyways. And here using a method called `isAfterAssessmentDueDate()` makes the code more readable (you don't need to spot the small exclamation mark)

The other methods in `ExerciseDateService` also all use the same format (e.g. `isBeforeLatestDueDate` and `isAfterLatestDueDate`):

![grafik](https://github.com/ls1intum/Artemis/assets/26540346/fb148144-fbcb-4ebb-b44c-39e935dabdd7)

Reply:
I see your point, but if we argue in favour of readability, we have to rename the method to `isAfterOrEqualToAssessmentDueDate()`. Because we can't expect a developer to see that there also is another method and that the method is actually not named quite right because it doesn't something different.

We don't have to pay for every character we use, and it's bad practice.

I would also suggest renaming the other methods you mentioned accordingly then to remove these bad practices too. Either you or I can do it in another PR. Let me know.

So, in the end, we can either just invert the call (!) or name the method properly. But the way it is right now, I can't approve it, sorry. I just don't agree with your assessment on this subject.